### ISSUE TITLE
train speed too slow 

### ISSUE THREAD

**Issue by Edwardmark** [deepinsight/insightface#291]
I try to finetune   LResNet50E-IR model using glint Asian dataset, I run the following command:  
CUDA_VISIBLE_DEVICES='0,1,2' python -u train_softmax.py \
                                --network r50 \
                                --loss-type 0 \
                                --lr 0.005 \
                                --per-batch-size 64 \
                                --data-dir /mntML/dongbin/datasets/faces_asian_112x112 \
                                --pretrained ../models/model-r50-am-lfw/model,0000 \
                                --prefix "$PREFIX"
I use mxnet_cu90, and I use 1080Ti x 3，the speed is so slow, the log is as follows:

gpu num: 3
num_layers 50
image_size [112, 112]
num_classes 93979
Called with argument: Namespace(batch_size=192, beta=1000.0, beta_freeze=0, beta_min=5.0, bn_mom=0.9, ckpt=1, ctx_num=3, cutoff=0, data_dir='/mntML/dongbin/datasets/faces_asian_112x112', easy_margin=0, emb_size=512, end_epoch=100000, fc7_wd_mult=1.0, gamma=0.12, image_channel=3, image_h=112, image_w=112, loss_type=0, lr=0.005, lr_steps='', margin=4, margin_a=1.0, margin_b=0.0, margin_m=0.5, margin_s=64.0, max_steps=0, mom=0.9, network='r50', num_classes=93979, num_layers=50, per_batch_size=64, power=1.0, prefix='../model-r50-asian/model-asian', pretrained='../models/model-r50-am-lfw/model,0000', rand_mirror=1, rescale_threshold=0, scale=0.9993, target='lfw,cfp_fp,agedb_30', use_deformable=0, verbose=2000, version_act='prelu', version_input=1, version_output='E', version_se=0, version_unit=3, wd=0.0005)
loading ['../models/model-r50-am-lfw/model', '0000']
[07:58:36] src/engine/engine.cc:55: MXNet start using engine: ThreadedEnginePerDevice
init resnet 50
0 1 E 3 prelu
INFO:root:loading recordio /mntML/dongbin/datasets/faces_asian_112x112/train.rec...
header0 label [2830147. 2924126.]
id2range 93979
2830146
rand_mirror 1
loading bin 0
loading bin 1000
loading bin 2000
loading bin 3000
loading bin 4000
loading bin 5000
loading bin 6000
loading bin 7000
loading bin 8000
loading bin 9000
loading bin 10000
loading bin 11000
(12000L, 3L, 112L, 112L)
ver lfw
loading bin 0
loading bin 1000
loading bin 2000
loading bin 3000
loading bin 4000
loading bin 5000
loading bin 6000
loading bin 7000
loading bin 8000
loading bin 9000
loading bin 10000
loading bin 11000
loading bin 12000
loading bin 13000
(14000L, 3L, 112L, 112L)
ver cfp_fp
loading bin 0
loading bin 1000
loading bin 2000
loading bin 3000
loading bin 4000
loading bin 5000
loading bin 6000
loading bin 7000
loading bin 8000
loading bin 9000
loading bin 10000
loading bin 11000
(12000L, 3L, 112L, 112L)
ver agedb_30
lr_steps [106666, 160000, 213333]
call reset()
/usr/local/lib/python2.7/dist-packages/mxnet/module/base_module.py:490: UserWarning: Optimizer created manually outside Module but rescale_grad is not normalized to 1.0/batch_size/num_workers (0.333333333333 vs. 0.00520833333333). Is this intended?
  optimizer_params=optimizer_params)
INFO:root:Epoch[0] Batch [20]	Speed: 29.62 samples/sec	acc=0.000000
INFO:root:Epoch[0] Batch [40]	Speed: 36.43 samples/sec	acc=0.006510
INFO:root:Epoch[0] Batch [60]	Speed: 41.90 samples/sec	acc=0.040104
INFO:root:Epoch[0] Batch [80]	Speed: 55.39 samples/sec	acc=0.102344
INFO:root:Epoch[0] Batch [100]	Speed: 76.50 samples/sec	acc=0.155469
INFO:root:Epoch[0] Batch [120]	Speed: 70.54 samples/sec	acc=0.182031
INFO:root:Epoch[0] Batch [140]	Speed: 72.65 samples/sec	acc=0.203125  

Anybody knows why?

**Comment by wuyxiquanquan** [comment]
......your acc is raising so fast.

**Comment by Edwardmark** [comment]
@wuyxiquanquan I fine-tune the model, so it's normal.

**Comment by Lane565** [comment]
@Edwardmark can you share your faces_asian_112x112 on BaiduYun?

**Comment by Edwardmark** [comment]
@Lane565 I downloaded from deepglint, they offer baiduyun link.
http://trillionpairs.deepglint.com/data

**Comment by Lane565** [comment]
I know that 90G jpg file , but i don't know how to get .rec and .idx and property

**Comment by Edwardmark** [comment]
follow the guide here:  
https://github.com/deepinsight/insightface/issues/256

**Comment by haoweiliang1996** [comment]
you should use a ssd  instead hdd to store the rec file

**Comment by haoweiliang1996** [comment]
when i use a raid hdd to train mobilefacenet with two gtx1080ti ,i get 200/s.when i train it in a samsung 960 pcie ssd ,i get 1200/s

**Comment by nttstar** [comment]
Compile mxnet from source may also help.

**Comment by ayush431** [comment]
testing speed is too slow for me?do you know whats the reason
itis taking time in extracting feature from a image.
